Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -48,9 +48,11 @@ public class ApolloCodegen {
         throw CodegenError.multipleFilesButNotDirectoryURL(folderURL)
       }
     case .singleFile(let fileURL):
-      guard fileURL.apollo.isSwiftFileURL else {
-        throw CodegenError.singleFileButNotSwiftFileURL(fileURL)
-      }
+      if options.codegenEngine == .typescript {
+        guard fileURL.apollo.isSwiftFileURL else {
+         throw CodegenError.singleFileButNotSwiftFileURL(fileURL)
+        }
+      } // else we're fine with JSON at this point for intermediate purposes.
```

Comment:
Once default codegen becomes swift codegen, this will fail the tests that check that a folder URL can't be passed in here.